UnityProject/Assets/Scripts/UI/Action/RequestGameAction.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections;
+using UnityEngine;
+using Utility = UnityEngine.Networking.Utility;
+using Mirror;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+public class RequestGameAction : ClientMessage
+{
+	public static short MessageType = (short)MessageTypes.RequestGameAction;
+
+	public int ComponentLocation;
+	public uint NetObject;
+	public Type ComponentType;
+
+	public static readonly Dictionary<ushort, Type> componentIDToComponentType = new Dictionary<ushort, Type>(); //These are useful
+	public static readonly Dictionary<Type, ushort> componentTypeToComponentID = new Dictionary<Type, ushort>();
+
+	static RequestGameAction()
+	{
+		//initialize id mappings
+		var alphabeticalComponentTypes =
+			typeof(IActionGUI).Assembly.GetTypes()
+				.Where(type => typeof(IActionGUI).IsAssignableFrom(type))
+				.OrderBy(type => type.FullName);
+		ushort i = 0;
+		foreach (var componentType in alphabeticalComponentTypes)
+		{
+			componentIDToComponentType.Add(i, componentType);
+			componentTypeToComponentID.Add(componentType, i);
+			i++;
+		}
+	
+	}
+
+
+
+
+
+
+
+
+	public override IEnumerator Process()
+	{
+		yield return WaitFor(NetObject);
+		//Logger.Log("ComponentLocation > " + ComponentLocation  + " NetworkObject > " +  NetworkObject + " ComponentType > " + ComponentType);
+		if (SentByPlayer != ConnectedPlayer.Invalid)
+		{
+			var IActionGUIs = NetworkObject.GetComponentsInChildren(ComponentType);
+			if ((IActionGUIs.Length > ComponentLocation)) {
+				var IServerActionGUI = (IActionGUIs[ComponentLocation] as IServerActionGUI);
+				IServerActionGUI.CallActionServer(SentByPlayer);
+			}
+		}
+	}
+
+
+	public static RequestGameAction Send(IServerActionGUI iServerActionGUI )
+	{
+		var netObject = iServerActionGUI.GetNetworkIdentity();
```

Comment:
The reason I decided to do this is because they could be calling it from component without a network behaviour in the game object, so if that was the case then this would fail, so It's a bit of extra logic so they can specify the correct network identity to use if there Isn't one on the same game object

Reply:
@Bod9001 it shouldn't fail in that case. Whether the component implementing the logic is a NetworkBehavior or not shouldn't affect this system - what matters is that the object itself has a NetworkIdentity. Pretty much all objects in the game have a NetworkIdentity, even if none of their components are NetworkBehaviors.